logstash-core/src/main/java/org/logstash/FileLockFactory.java

```diff
@@ -39,7 +41,8 @@
 
     private FileLockFactory() {}
 
-    private static final Set<String> LOCK_HELD = Collections.synchronizedSet(new HashSet<String>());
+    private static final Set<String> LOCK_HELD = Collections.synchronizedSet(new HashSet<>());
```

Comment:
right, but I am not fully understanding why it is the factories job to manage the release of the lock.

FileLocks are closeable, so once you're done locking, you just close the lock and its over. Maybe it makes more sense to simply check whether the lock is both held and closed, and if it is, remove it from `LOCK_HELD` and proceed in obtaining the lock.

But yeah, I am just shooting in the dark here since I don't have context into how it is being used. If everything I say doesn't apply, then ignore me.